lib/neatjson.rb

```diff
@@ -115,16 +116,17 @@ def self.neat_generate(object,opts={})
 										one_line
 									end
 								end
-								keyvals.join(",\n") << opad << "}"
+								keyvals.join(",\n") + opad + "}"
 							else
 								keyvals = o.map{ |k,v| ["#{indent}#{opts[:indent]}#{k.to_s.inspect}",v] }
 								keyvals = keyvals.sort_by(&:first) if opts[:sorted]
 								if opts[:aligned]
 									longest = keyvals.map(&:first).map(&:length).max
-									keyvals.each{ |k,v| k.replace( "%-#{longest}s" % k ) }
+									#keyvals.each{ |k,v| k.replace( "%-#{longest}s" % k ) }
+									keyvals = keyvals.inject({}){|r, e| r["%-#{longest}s" % e.first] = e.last; r}
 								end
 								indent2 = "#{indent}#{opts[:indent]}"
-								keyvals.map! do |k,v|

```

Comment:
When I started with this, I reworked every mutable operation, without thinking too much.